src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -136,6 +138,26 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
         }
   }
 
+  override def simulateTransaction(stx: SignedTransaction, blockHeader: BlockHeader, storages: BlockchainStorages, validators: Validators): TxResult = {
```

Comment:
I'm wondering, should `updateSenderAccountBeforeExecution` be called here as in `Ledger.executeTransaction`?